test/Semantic/Spec.hs

```diff
@@ -24,5 +25,22 @@ spec = parallel $ do
     it "renders with the specified renderer" $ do
       output <- fmap runBuilder . runTaskOrDie $ parseTermBuilder TermSExpression [methodsBlob]
       output `shouldBe` "(Statements\n  (Method\n    (Empty)\n    (Identifier)\n    (Statements)))\n"
+
+  describe "gitParsing" $ do
+    it "parses a git output string" $ do
+      let input = "100644 tree ThisIsTheOid\t/this/is/the/path"
+      let expected = TreeEntry NormalMode TreeObject (OID "ThisIsTheOid") "/this/is/the/path"
+      parseEntry input `shouldBe` expected
+
+    it "parses nonsense into a default value" $ do
+      let input = "iel jgh\nf2 8i4p\r8f2y4fpoxin u3y2 unz"
+      let expected = TreeEntry OtherMode OtherObjectType (OID mempty) mempty
+      parseEntry input `shouldBe` expected
+
+    it "parses many outputs separated by \\NUL" $ do
```

Comment:
I’d also like to see an integration test that used `git` invocations in a temporary directory, but I think I’ll take care of that in a separate PR.

Reply:
Turns out I already did this and completely forgot about it.